pterodactyl.php

```diff
@@ -248,6 +248,14 @@ public function addService(
             }
 
             $meta['server_id'] = $pterodactyl_server->attributes->id;
+            if (isset($pterodactyl_server->attributes->relationships)
+                && isset($pterodactyl_server->attributes->relationships->allocations)
+                && isset($pterodactyl_server->attributes->relationships->allocations->data[0])
+            ) {
+                $allocation = $pterodactyl_server->attributes->relationships->allocations->data[0];
```

Comment:
It looks like you check all the properties exist for the relationship's allocation data, but then you assume that there is an `attributes` property that has both an `ip` and `port`. I would probably check that those properties exist before setting meta data, otherwise default to null. You could also add those checks on the `editService` method

Reply:
Updated